src/aws/roles.tf

```diff
@@ -309,3 +309,184 @@ resource "aws_iam_role_policy_attachment" "lambda_migrate_lambda_invoke_policy_a
+  name        = "${terraform.workspace}-marsha-medialive-custom-policy"
+  path        = "/"
+  description = "IAM policy needed access SSM in read-only mode"
+
+  policy = <<EOF
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Action": [
+                "s3:ListBucket",
+                "s3:PutObject",
+                "s3:GetObject",
+                "s3:DeleteObject"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Effect": "Allow",
+            "Action": [
+                "mediastore:ListContainers",
+                "mediastore:PutObject",
+                "mediastore:GetObject",
+                "mediastore:DeleteObject",
+                "mediastore:DescribeObject"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Effect": "Allow",
+            "Action": [
+                "logs:CreateLogGroup",
+                "logs:CreateLogStream",
+                "logs:PutLogEvents",
+                "logs:DescribeLogStreams",
+                "logs:DescribeLogGroups"
+            ],
+            "Resource": "arn:aws:logs:*:*:*"
+        },
+        {
+            "Effect": "Allow",
+            "Action": [
+                "mediaconnect:ManagedDescribeFlow",
+                "mediaconnect:ManagedAddOutput",
+                "mediaconnect:ManagedRemoveOutput"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Effect": "Allow",
+            "Action": [
+                "ec2:describeSubnets",
+                "ec2:describeNetworkInterfaces",
+                "ec2:createNetworkInterface",
+                "ec2:createNetworkInterfacePermission",
+                "ec2:deleteNetworkInterface",
+                "ec2:deleteNetworkInterfacePermission",
+                "ec2:describeSecurityGroups"
+            ],
+            "Resource": "*"
+        },
+        {
+            "Effect": "Allow",
+            "Action": [
+                "mediapackage:DescribeChannel"
+            ],
+            "Resource": "*"
+        }
+    ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy_attachment" "medialive_custom_policiy_attachment" {
+  role        = "${aws_iam_role.medialive_access_role.name}"
+  policy_arn  = "${aws_iam_policy.medialive_custom_policy.arn}"
+}
+
+
+resource "aws_iam_role" "lambda_medialive_invocation_role" {
+  name = "${terraform.workspace}-marsha-lambda-medialive-invocation-role"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "lambda.amazonaws.com"
+      },
+      "Effect": "Allow"
+    }
+  ]
+}
+EOF
+}
+
+
+resource "aws_iam_role_policy_attachment" "lambda_medialive_logging_policy_attachment" {
+  role       = "${aws_iam_role.lambda_medialive_invocation_role.name}"
+  policy_arn = "${aws_iam_policy.lambda_logging_policy.arn}"
```

Comment:
> Also you should add an issue to uniformize invocation roles because the first lambdas we wrote share the same role but the new lambdas have a specific role. The old lambdas should be refactored to the new way of doing things?

yes definitely